src/main/java/com/celements/payment/service/ComputopService.java

```diff
@@ -92,8 +91,126 @@ String hashPaymentData(@NotNull String paymentData) {
     return "";
   }
 
+  @Override
+  public Map<String, String> encryptPaymentData(String transactionId, String orderDescription,
+      BigDecimal amount, String currency) {
+    String dataPlainText = getPaymentDataPlainString(transactionId, orderDescription, amount,
+        currency);
+    Map<String, String> encryptedData = new HashMap<>();
+    encryptedData.put(FORM_INPUT_NAME_LENGTH, Integer.toString(dataPlainText.length()));
+    String encyrptedData = encryptString(dataPlainText.getBytes(), getBlowfishKey());
+    if (encyrptedData != null) {
+      encryptedData.put(FORM_INPUT_NAME_DATA, encyrptedData);
+    }
+    return encryptedData;
+  }
+
+  @Override
+  public Map<String, String> decryptCallbackData(String encryptedCallback, int plainDataLength) {
+    checkNotNull(encryptedCallback);
+    byte[] decryptedData = decryptString(encryptedCallback, plainDataLength, getBlowfishKey());
+    Map<String, String> callbackData = new HashMap<>();
+    if (decryptedData != null) {
+      for (String parameter : (new String(decryptedData)).split("&")) {
+        String[] keyValue = parameter.split("=", 2);
```

Comment:
using Guavas `Splitter` returns Iterable and is perhaps more elegant/readable than working with indices/length